### DISCUSSION TITLE
DEMO: How to setup and run your first Templater JS "script"

### DISCUSSION THREAD

**Discussion by GitMurf** [SilentVoid13/Templater#187] [Templates Showcase]
Ok so the question of "how do I setup my first Templater script" comes up a TON when folks are trying to use Templater for the first time, especially when trying to copy one of the Templater scripts that someone posts to the community. I recorded this Demo video that hopefully will clear that up and show everyone from A-Z the process to get their first Templater JS script working. @SilentVoid13 feel free to add this video to your GH readme. I also posted in Obsidian Discord server in the Plugins channel.

https://user-images.githubusercontent.com/64155612/116945646-10c84c00-ac2d-11eb-864c-4744ba90e860.mp4

**Comment by welpdx** [comment]
Tip: When copying html text over to Obsidian, it [adds unnecessary slashes and removes tabs](https://forum.obsidian.md/t/obsidian-adds-unnecessary-slashes-to-my-code-block-removes-all-tabs-t/11061). These slashes breaks the code. To avoid this,  [use Ctrl + Shift + V when pasting](https://forum.obsidian.md/t/obsidian-adds-unnecessary-slashes-to-my-code-block-removes-all-tabs-t/11061/2).

**Reply by GitMurf** [reply]
Yes, see 2:35 in my video above. I demonstrate the same thing :)

**Reply by welpdx** [reply]
Oh I must've missed it. And after I wrote that above comment, I actually saw that you have written the same thing [somewhere else](https://gist.github.com/GitMurf/c0500d8bcef2a0f83c0b526cf07fc265#gistcomment-3716071) already. 

Sorry for repeating the same thing.  XD

**Comment by jroachgit** [comment]
So cool! I'm keen to hear or read about different use cases that are in use. Are they documented / shared some where?

**Reply by welpdx** [reply]
Yes, there are tons of user made scripts and templates in this github's [discussions page](https://github.com/SilentVoid13/Templater/discussions). 

glhf!

**Comment by dissyllabic** [comment]
Thanks man! great plugin, great tutorial. Appreciated.

**Comment by jesusyesares** [comment]
Hi @GitMurf,

there's any problem with the video. Could you restore it? 

Thanks.

<img width="545" alt="imagen" src="https://user-images.githubusercontent.com/2352123/202870935-5de72ffc-6363-476c-9c36-641ad2ca4f0a.png">

**Reply by gnapse** [reply]
> Hi @GitMurf,
> 
> there's any problem with the video. Could you restore it?

I experience the same issue, and found a workaround: right-click the video and select "Copy video link" (option name in Firefox, it may be labelled differently in other browsers). Then visit the copied URL directly in a new tab. I'm pasting the URL below for convenience:

```
https://user-images.githubusercontent.com/64155612/116945646-10c84c00-ac2d-11eb-864c-4744ba90e860.mp4
```

**Reply by gnapse** [reply]
Oh, and now I see it works as an embedded video when I paste it right here 😮

https://user-images.githubusercontent.com/64155612/116945646-10c84c00-ac2d-11eb-864c-4744ba90e860.mp4

**Comment by sanjarcode** [comment]
Where is the code? Please add a link.

**Comment by nprime496** [comment]


**Reply by dissyllabic** [reply]
I think the resources linked to on the main GitHub Templater "landing page" gives you excellent resources for learning templater. 
## Quick link
```groovy
@chhoumann Templates showcase: https://github.com/chhoumann/Templater_Templates
@zachatoo Templates showcase: https://zachyoung.dev/posts/templater-snippets
@lguenth Templates showcase: https://github.com/lguenth/obsidian-templates
@tallguyjenks video: https://youtu.be/2234DXKbNgM?t=1944](https://youtu.be/2234DXKbNgM?t=1944
``` 
**And I think this video works well as part of the the resources above**

**Comment by Emotion04** [comment]
Thanks guy! great plugin, great tutorial. Appreciated.

**Comment by Jay-1111** [comment]
please accept my apologies, I am very new to this and would appreciate patience with me. I know this is an elementary question

I have done what shows on the video, I have added the Java Script using crtl+shit+V and the readview is still showing script

**Reply by Zachatoo** [reply]
Templater doesn't dynamically show content when switching between editing/reading mode. You have to execute Templater, usually using a command/hotkey, which will run your script and permanently replace your script with the output.

**Reply by Jay-1111** [reply]
yes I have done this.  it still only shows script

**Reply by Zachatoo** [reply]
Which specific command are you using? Are there any errors that show up after trying to execute Templater?

To check for errors, first open the developer tools by pressing <kbd>Ctrl</kbd><kbd>Shift</kbd><kbd>I</kbd> on Windows/Linux or <kbd>Cmd</kbd><kbd>Opt</kbd><kbd>I</kbd> on macOS, and then go to the "Console" tab.

**Reply by Jay-1111** [reply]
thanks I will have a go.  I appreciate your help.

**Comment by cassgraz24-ctrl** [comment]
This
